lib/broker/rabbit/queue.ts

```diff
@@ -81,17 +82,15 @@ export class Queue {
 	 * @returns {string} string
 	 */
 	private queueName(routingKey: string): string {
-		const splitEventName = routingKey.split('.');
+		const splitEventName = routingKey.toLowerCase().split('.');
 
-		if (![3, 4].includes(splitEventName.length))
-			throw new Error(
-				`invalid event name: "${routingKey}" length, should match resource.origin.action or resource.origin.action.dest pattern`,
-			);
+		if (![3, 4, 5].includes(splitEventName.length))
+			throw new MissingAttributeError(INVALID_ROUTING_KEY);
 
 		const last = splitEventName.length - 1;
 		if (splitEventName[last] !== 'all')
-			return `${this.config.APP_NAME.toLocaleLowerCase()}-${routingKey.toLocaleLowerCase()}.all`;
+			return `${this.config.APP_NAME.toLowerCase()}-${routingKey}.all`;
```

Comment:
What if the event is 4 words long? You'd be adding a fifth word `all`?

Also, you could only set the suffix in the condition given the return is basically the same for both conditions.